my_enumerable.rb

```diff
@@ -1,85 +1,167 @@
+      checker.empty? ? (return false) : (return true)
+    end
+
+    return true unless all? { |x| !yield(x) }
 
-    !my_all? { |x| !yield(x) }
+    false
   end
 
-  def my_none?
-    return to_enum(_method_) unless block_given?
+  def my_none?(*type)
+    origin = to_a
+    return true if origin == []
+
+    if !block_given? and type.length.zero?
+      return true if origin.my_all? { |x| x.nil? || x == false }
+
+      return false
+    end
+
+    if type.length == 1
+      checker = origin.my_select { |x| x =~ type[0] } if type[0].instance_of?(Regexp)
+      checker = origin.my_select { |x| x == type[0] } unless type[0].instance_of?(Regexp)
+      checker.empty? ? (return true) : (return false)
+    end
 
     !my_any? { |x| yield(x) }
   end
 
   def my_count(argument = nil)
+    origin = to_a
+
+    return origin.length if !block_given? and !argument
+
     if argument
-      count = my_select { |x| x == argument }
+      count = origin.my_select { |x| x == argument }
       return count.length
 
     elsif block_given?
-      count = my_select { |x| yield(x) }
+      count = origin.my_select { |x| yield(x) }
       return count.length
     end
 
     length
   end
 
-  def my_map(proc = nil)
-    return to_enum(_method_) unless block_given?
+  def my_map(my_proc = nil)
+    return to_enum(__method__) unless block_given?
 
+    origin = to_a
     array = []
-    if proc
-      my_each { |x| array << procedure.call(x) }
+    if my_proc
+      origin.my_each { |x| array << my_proc.call(x) }
+      return array
     else
-      my_each { |x| array << yield(x) }
+      origin.my_each { |x| array << yield(x) }
     end
     array
   end
 
-  def my_inject(number = nil, operator = nil)
-    if !number.nil?
-      operator = number
-      number = nil
-      to_a.my_each { |item| number = number.nil? ? item : number.send(operator, item) }
-    elsif !operator.nil?
-      to_a.my_each { |item| number = number.nil? ? item : number.send(operator, item) }
-    else
-      to_a.my_each { |item| number = number.nil? ? item : yield(number, item) }
+  def my_inject(*arg)
+    origin = to_a
+    if !block_given? && arg.length == 1 && origin != []
+      number = origin[0]
+      origin.my_each do |item, i|
+        number = number.send(arg[0], item) if i.positive?
+      end
+      return number
+    end
+
+    if !block_given? && arg.length == 2 && origin != []
+      number = arg[0]
+      origin.my_each do |item|
+        number = number.send(arg[1], item)
+      end
+      return number
+    end
+
+    if block_given? && arg.length == 1 && origin != []
+      number = arg[0]
+      origin.my_each do |item|
+        number = yield(number, item)
+      end
+      return number
+    end
+
+    if block_given? && arg.length.zero? && origin != []
+      number = origin[0]
+      origin.my_each_with_index do |item, i|
+        number = yield(number, item) if i.positive?
+      end
     end
     number
   end
```

Comment:
- [ ]  `my_inject` should **raise** a` localJumpError` when no block or argument is given.
```ruby
array = [1,2,3,4,5]
array.my_inject
```
_Hint: Yours currently returns `nil` 👇 when I run the code above_
![Screen Shot 2020-10-04 at 9 35 41 PM](https://user-images.githubusercontent.com/48804258/95025371-8f560e80-0689-11eb-8ddd-09c1b8d9a834.png)

- [ ] When a symbol is given without an initial value,  and `my_inject` is called on an array, it should combine each element of the collection by applying the symbol as a named method
```ruby 
array = [1,2,3,4,5]
p array.inject(:*) => 120
p array.my_inject(:*) 
```
_Yours throws a `NoMethodError: undefined method 'positive?' for nil:NilClass`. Screenshot of output from terminal below 👇_
![Screen Shot 2020-10-04 at 9 42 31 PM](https://user-images.githubusercontent.com/48804258/95025517-831e8100-068a-11eb-897b-9dbce5133c58.png)

 
- [ ] When a symbol is given without an initial value,  and `my_inject` is called on a range, it should combine each element of the collection by applying the symbol as a named method
```ruby
range = Range.new(5,50)
p range.inject(:*) 
p range.my_inject(:*) 
```
_Yours throws a `NoMethodError: undefined method 'positive?' for nil:NilClass`. Screenshot of output from terminal below 👇_

![Screen Shot 2020-10-04 at 9 47 11 PM](https://user-images.githubusercontent.com/48804258/95025594-2a031d00-068b-11eb-9cb0-245cf5abba1e.png)